--- a/drivers/gpu/drm/bridge/analogix/analogix_dp_reg.h
+++ b/drivers/gpu/drm/bridge/analogix/analogix_dp_reg.h
@@ -15,9 +15,27 @@
 #define ANALOGIX_DP_VIDEO_CTL_1			0x20
 #define ANALOGIX_DP_VIDEO_CTL_2			0x24
 #define ANALOGIX_DP_VIDEO_CTL_3			0x28
+#define ANALOGIX_DP_VIDEO_CTL_4			0x2C
 
 #define ANALOGIX_DP_VIDEO_CTL_8			0x3C
 #define ANALOGIX_DP_VIDEO_CTL_10		0x44
+#define ANALOGIX_DP_TOTAL_LINE_CFG_L		0x48
+#define ANALOGIX_DP_TOTAL_LINE_CFG_H		0x4C
+#define ANALOGIX_DP_ACTIVE_LINE_CFG_L		0x50
+#define ANALOGIX_DP_ACTIVE_LINE_CFG_H		0x54
+#define ANALOGIX_DP_V_F_PORCH_CFG		0x58
+#define ANALOGIX_DP_V_SYNC_WIDTH_CFG		0x5C
+#define ANALOGIX_DP_V_B_PORCH_CFG		0x60
+#define ANALOGIX_DP_TOTAL_PIXEL_CFG_L		0x64
+#define ANALOGIX_DP_TOTAL_PIXEL_CFG_H		0x68
+#define ANALOGIX_DP_ACTIVE_PIXEL_CFG_L		0x6C
+#define ANALOGIX_DP_ACTIVE_PIXEL_CFG_H		0x70
+#define ANALOGIX_DP_H_F_PORCH_CFG_L		0x74
+#define ANALOGIX_DP_H_F_PORCH_CFG_H		0x78
+#define ANALOGIX_DP_H_SYNC_CFG_L		0x7C
+#define ANALOGIX_DP_H_SYNC_CFG_H		0x80
+#define ANALOGIX_DP_H_B_PORCH_CFG_L		0x84
+#define ANALOGIX_DP_H_B_PORCH_CFG_H		0x88
 
 #define ANALOGIX_DP_SPDIF_AUDIO_CTL_0		0xD8
 
@@ -75,6 +93,7 @@
 #define ANALOGIX_DP_HDCP_CTL			0x648
 
 #define ANALOGIX_DP_LINK_BW_SET			0x680
+#define ANALOGIX_DP_LINK_POLICY			0x9D8
 #define ANALOGIX_DP_LANE_COUNT_SET		0x684
 #define ANALOGIX_DP_TRAINING_PTN_SET		0x688
 #define ANALOGIX_DP_LN0_LINK_TRAINING_CTL	0x68C
@@ -177,6 +196,9 @@
 
 /* ANALOGIX_DP_VIDEO_CTL_10 */
 #define FORMAT_SEL				(0x1 << 4)
+
+/* ANALOGIX_DP_VIDEO_CTL_4 */
+#define BIST_EN					(0x1 << 3)
 #define INTERACE_SCAN_CFG			(0x1 << 2)
 #define VSYNC_POLARITY_CFG			(0x1 << 1)
 #define HSYNC_POLARITY_CFG			(0x1 << 0)
@@ -283,6 +305,25 @@
 #define INT_POL1				(0x1 << 1)
 #define INT_POL0				(0x1 << 0)
 
+/* Video timing cfg (REGISTER mode) */
+#define TOTAL_LINE_CFG_L(x)			(((x) & 0xff) << 0)
+#define TOTAL_LINE_CFG_H(x)			(((x) & 0xf) << 0)
+#define ACTIVE_LINE_CFG_L(x)			(((x) & 0xff) << 0)
+#define ACTIVE_LINE_CFG_H(x)			(((x) & 0xf) << 0)
+#define V_F_PORCH_CFG(x)			(((x) & 0xff) << 0)
+#define V_SYNC_WIDTH_CFG(x)			(((x) & 0xff) << 0)
+#define V_B_PORCH_CFG(x)			(((x) & 0xff) << 0)
+#define TOTAL_PIXEL_CFG_L(x)			(((x) & 0xff) << 0)
+#define TOTAL_PIXEL_CFG_H(x)			(((x) & 0x3f) << 0)
+#define ACTIVE_PIXEL_CFG_L(x)			(((x) & 0xff) << 0)
+#define ACTIVE_PIXEL_CFG_H(x)			(((x) & 0x3f) << 0)
+#define H_F_PORCH_CFG_L(x)			(((x) & 0xff) << 0)
+#define H_F_PORCH_CFG_H(x)			(((x) & 0xf) << 0)
+#define H_SYNC_CFG_L(x)				(((x) & 0xff) << 0)
+#define H_SYNC_CFG_H(x)				(((x) & 0xf) << 0)
+#define H_B_PORCH_CFG_L(x)			(((x) & 0xff) << 0)
+#define H_B_PORCH_CFG_H(x)			(((x) & 0xf) << 0)
+
 /* ANALOGIX_DP_SYS_CTL_1 */
 #define DET_STA					(0x1 << 2)
 #define FORCE_DET				(0x1 << 1)
@@ -424,3 +465,6 @@
 #define PSR_VID_CRC_ENABLE			(0x1 << 0)
 
 #endif /* _ANALOGIX_DP_REG_H */
+
+/* ANALOGIX_DP_LINK_POLICY */
+#define ALTERNATE_SR_ENABLE			(0x1 << 7)
